logbook-parent/pom.xml

```diff
@@ -317,14 +317,17 @@
         <dependency>
             <groupId>com.google.code.findbugs</groupId>
             <artifactId>jsr305</artifactId>
+            <scope>provided</scope>
```

Comment:
@kasmarian I have done a [detailed analysis](https://dev.to/msdousti/how-gradle-disagreed-with-our-maven-project-3n4a), would love to get feedback!

The issue will happen if **all** three conditions are satisifed (and only in Gradle-based projects):

1. Logbook Parent module has specified a scope like `provided` or `test` for a dependency X in the `dependencyManagement` section
2. Logbook Parent module has imported dependency X in the `dependencies` section, **without** specifying the scope.
3. A child module has a `dependencyManagement` section, that imports a BOM dependency (`scope = import`). This BOM must have dependency X in its own `dependencyManagement` section, without specifying the `scope`.


In our example, `Lombok` was (1) included in the parent both in dependency management section (`scope = provided`) and (2) in the dependency section (no scope). Then (3) I imported `spring-boot-dependencies` as a BOM, which had no scope.

For `ktor-client-cio-jvm` and `jakarta.servlet-api`, this does not happen as at least condition 2 does not hold: They are not included in the `dependencies` section of the parent.

Reply:
That's an interesting finding and a nice article! (maybe you could also add a link to the GH repo with your playground project in the article)

> However, if a child module uses them in the dependencies section, and provides no scope, and condition 3 holds, it can also create a compile-time transitive dependency.

If someone defines a dependency and doesn't specify a scope, I'd expect them to want a compile-time dependency. Then we probably should be fine here. Would the dependency scope from `dependencies` take priority in case someone specifies another scope?

Reply:
> maybe you could also add a link to the GH repo with your playground project in the article
Thanks. Good suggestion, will do!

> I'd expect them to want a compile-time dependency. Then we probably should be fine here.
I expect the same.

> Would the dependency scope from dependencies take priority in case someone specifies another scope?
I guess so, because that's what I'm doing in this PR: In the parent project, I specified the scope `provided` for Lombok. Even though `spring-boot-dependencies` does not specify the scope (effectively making it compile), the explicit scope wins (both for Maven and Gradle).

I also tested this guess:

![image](https://github.com/zalando/logbook/assets/3616518/7ec6d87e-b1a2-485d-a810-0fb5f7c93543)

which, as expected, results in the scope explicitly mentioned in the `dependencies` section (i.e., `test`):

![image](https://github.com/zalando/logbook/assets/3616518/b0e45d2d-ff53-44dc-b8cd-5afb8ce27538)

Reply:
Thanks for checking. Looks good then. One last question: how can we make sure not to have a regression here? Maybe a comment explaining why the scope is needed here would help?

Reply:
Done in 18bea53